cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -362,15 +386,16 @@ public Void call() throws Exception {
       }
     }
 
-    String extraClassPath = "$PWD/" + CDAP_SPARK_JAR + "/lib/*";
+    String extraClassPath = "$PWD/" + CDAP_LAUNCHER_JAR + File.pathSeparator + "$PWD/" + CDAP_SPARK_JAR + "/lib/*";
```

Comment:
is ``CDAP_LAUNCHER_JAR`` a directory? Also, should we use ``Joiner.on(File.pathSeparator)`` here instead of using the forward slashes